### PR TITLE
Migrate music_test.js

### PR DESCRIPTION
**Pull request by rvilarl**

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -24,10 +24,10 @@ module.exports = (grunt) => {
   const TARGET_TESTS = path.join(BUILD_DIR, 'vexflow-tests.js');
   const TEST_SOURCES = ['tests/vexflow_test_helpers.js', 'tests/mocks.js', 'tests/*_tests.js', 'tests/run.js'];
 
-  function webpackConfig(target, preset, mode) {
+  function webpackConfig(target, entry, preset, mode) {
```
Comment (wassertim) [inline Gruntfile.js:27]: is preset still used?

File: Gruntfile.js [hunk 2]
```diff
@@ -68,8 +68,9 @@ module.exports = (grunt) => {
     };
   }
 
-  const webpackProd = webpackConfig(TARGET_MIN, ['@babel/preset-env'], 'production');
-  const webpackDev = webpackConfig(TARGET_RAW, ['@babel/preset-env'], 'development');
+  const webpackProd = webpackConfig(TARGET_MIN, MODULE_ENTRY, ['@babel/preset-env'], 'production');
```
Comment (wassertim) [inline Gruntfile.js:71]: we should remove babel, we don't use it anymore, we use typescript

File: Gruntfile.js [hunk 3]
```diff
@@ -68,8 +68,9 @@ module.exports = (grunt) => {
...
+  const webpackDev = webpackConfig(TARGET_RAW, MODULE_ENTRY, ['@babel/preset-env'], 'development');
+  const webpackTest = webpackConfig('vexflow-tests-ts.js', path.join(BASE_DIR, 'tests/index.ts'), ['@babel/preset-env'], 'development');
```
Comment (wassertim) [inline Gruntfile.js:73]: why does the 1st parameter have this strange name?

File: Gruntfile.js [hunk 4]
```diff
@@ -98,7 +100,7 @@ module.exports = (grunt) => {
       options: { fix: true },
     },
     qunit: {
-      files: ['tests/flow.html'],
+      files: ['tests/flow.html', 'tests/flow_ts.html'],
```
Comment (wassertim) [inline Gruntfile.js:103]: why is new flow_ts needed?

**Review by wassertim — CHANGES_REQUESTED** [review]

File: Gruntfile.js (hunk 1 above)
Comment (rvilarl) [reply Gruntfile.js:27]: It is used but I do not know what for

File: Gruntfile.js (hunk 2 above)
Comment (rvilarl) [reply Gruntfile.js:71]: Well I wanted to leave the javascript part as untouched as possible

File: Gruntfile.js (hunk 4 above)
Comment (rvilarl) [reply Gruntfile.js:103]: In order to check the ts tests, as discussed I decided to leave js as it is in order to keep  the number of changes to a minimum

File: Gruntfile.js (hunk 3 above)
Comment (rvilarl) [reply Gruntfile.js:73]: I left the constant names untouched. vexflow-tests-ts.js is the compiled version of the ts tests. Please note that this is vexflow-tests.js for the js tests.

**Comment by rvilarl** [comment]
@wassertim I suggest not to spend too much trying to bring both tests together. I mean js and ts tests.

This is an itemediate step and there is not much difference in the output and you still get a summary:

image

**Comment by rvilarl** [comment]
This also gives a view on the current progress! :)

File: Gruntfile.js (hunk 1 above)
Comment (rvilarl) [reply Gruntfile.js]: It was the preset environment for babel. No longer needed and removed

File: Gruntfile.js (hunk 2 above)
Comment (rvilarl) [reply Gruntfile.js]: Babel removed.

**Pull request closed without merging**